### DISCUSSION TITLE
the `auto` pipeline layout option.

### DISCUSSION THREAD

**Discussion by mobile-bungalow** [gfx-rs/wgpu#6469] [Q&A] [answered]
the web gpu spec as described on MDN includes [an `auto` option](https://developer.mozilla.org/en-US/docs/Web/API/GPUDevice/createRenderPipeline#layout) when constructing a pipeline. This is quite useful and really cuts down on boiler plate and the dual FFI problem. Would `wgpu` ever consider generating a layout descriptor from a shader source module using naga?

**Comment by mobile-bungalow** [comment]
here is [an example](https://github.com/mobile-bungalow/Kinnara/blob/master/src/bind_group/mod.rs) (one of a couple i've made) doing this for compute shaders. ignore the `directives` modifiers.

**Comment by cwfitzgerald** [comment, marked as answer]
Setting https://docs.rs/wgpu/latest/wgpu/struct.RenderPipelineDescriptor.html#structfield.layout to None, is equivalent to "auto" in WebGPU.

**Reply by mobile-bungalow** [reply]
Okay, that's really good to know. I was confused because the documentation doesn't mention this behavior and the field was required in previous versions of wgpu.
